### PR TITLE
WIP: Feat/add use apollo

### PR DESCRIPTION
**Pull request by KevinDanikowski**
1. added apollo and example queries and components
2. added means of environmental set ups and env variables

### CODE REVIEW

**Comment by KevinDanikowski** [comment]
@amclin 
With respect to the other PRs which are wrapped up in this PR:
1. For PageTitle you said "Is the key really necessary? I don't think eslint-react nor react is going to issue a warning here since it's wrapped in the element" but the key is actually specific for React Helmet which is baked into nextjs Head.
2. What do you mean by "Please revert these and use comments for single-line excludes", I've added eslint-disable comments but some times I thought it best to add the eslint "warn" setting in the .eslintrc . 
3. The bablerc file change is for fix for #127  
4. I will look into adding tests for _document.js, i'd prefer to do that in a test PR i'll have to do after this to add in react-testing-library  if this is ok.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+**/*.test.js
+**/*.test.jsx
```
Comment (amclin) [inline .eslintignore]: Can you revert this? It would be best to keep coding expectations consistent across all files.

File: .eslintrc.json [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-        "react"
-    ],
-    "rules": {
-      "react/prop-types": ["error"]
-    }
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "plugins": ["react"],
+  "rules": {
```
Comment (amclin) [inline .eslintrc.json]: Can we revert these?

File: helpers/copy-template-files.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const cpy = require('cpy')
```
Comment (amclin) [inline helpers/copy-template-files.js]: Generally if we do significant refactoring like this, it should be a separate change/PR from feature enhancements. That makes it easier to test, review, revert, and/or mitigate issues.

File: package.json [hunk 4]
```diff
@@ -89,6 +89,11 @@
     ],
     "coveragePathIgnorePatterns": [
       "<rootDir>/build/",
+      "<rootDir>/helpers/",
```
Comment (amclin) [inline package.json]: These shouldn't be ignored

File: templates/default/eslintignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+**/*.test.js
```
Comment (amclin) [inline templates/default/eslintignore]: revert this please, test files should have code linting consistent with functional files

File: templates/default/eslintrc.json [hunk 6]
```diff
@@ -19,6 +19,9 @@
   },
   "plugins": ["react"],
   "rules": {
-    "react/prop-types": ["error"]
+    "react/prop-types": ["error"],
```
Comment (amclin) [inline templates/default/eslintrc.json]: Let's review specifics please and see if these rule changes are actually necessary.

File: templates/default/src/__tests__/pages/example/second-page/__snapshots__/index.test.jsx.snap [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
-      href="/"
-      onClick={[Function]}
-      onMouseEnter={[Function]}
-    >
-      homepage
-    </a>
-  </p>
-</main>
+<div
+  style={
```
Comment (amclin) [inline templates/default/src/__tests__/pages/example/second-page/__snapshots__/index.test.jsx.snap]: If we're using NextJS styled-jsx, then we shouldn't be getting inline CSS like this

File: templates/default/src/components/atoms/Country/Country.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import React from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import PropTypes from 'prop-types'
+import COUNTRY_QUERY from '../../../graphql/queries/country'
+
+const Country = ({code}) => {
+  const { loading, error, data } = useQuery(COUNTRY_QUERY, {
+    variables: { code },
+  });
+  const wrapperStyle = {
```
Comment (amclin) [inline templates/default/src/components/atoms/Country/Country.jsx]: Can you look at the internal version of this project and follow the pattern of using NextJS styled-jsx please

File: templates/default/src/components/organisms/CountrySelector/CountrySelector.jsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  } if (loading) {
+    return <div>Loading Countries...</div>
+  } 
+
+  const {
+    countries = []
+  } = data
+
+  return (
+    <div style={{border: 'solid 1px black', margin: '15px', padding: '15px'}}>
```
Comment (amclin) [inline templates/default/src/components/organisms/CountrySelector/CountrySelector.jsx]: please don't use inline styles

File: templates/default/src/config/urls.js [hunk 10]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (amclin) [inline templates/default/src/config/urls.js]: Let's review this - I don't think config definitions belong in /src/

File: package.json (hunk 4 above)
Comment (KevinDanikowski) [reply package.json]: @amclin all of them are tested with the full build test, but it doesn't seem to be considered covered. it acts as if is not covered even though it is, like the `npm run test` is run in two different windows almost.

File: helpers/copy-template-files.js (hunk 3 above)
Comment (KevinDanikowski) [reply helpers/copy-template-files.js]: @amclin would you like a separate PR for this? If so, could we make sure we get it through the pipeline quickly. I don't mind doing a separate PR.

Comment (KevinDanikowski) [reply helpers/copy-template-files.js]: @amclin  see https://github.com/amclin/react-project-boilerplate/pull/160/files

File: package.json (hunk 4 above)
Comment (amclin) [reply package.json]: For the root project, I'd rather bump the coverage thresholds. Notice some helper files have tests to ensure coverage, and not relying on the integration test.

See #28

**Comment by amclin** [comment]
Duplicates #186

**Pull request closed without merging**